### ISSUE TITLE
Device service can't receive the system event by Redis Pub/Sub

### ISSUE THREAD

**Issue by weichou1229** [edgexfoundry/device-sdk-go#1366]
# 🐞 Bug Report

### Affected Services [**REQUIRED**]

 The issue is located in: device-sdk-go

### Is this a regression?

 No

### Description and Minimal Reproduction [**REQUIRED**]
Device service can't receive the system event by Redis Pub/Sub. ( I verified it works with MQTT )

- Run core-metadata and device-virtual
- Update the device service adminState from UNLOCKED to LOCKED
- core-metadata publish the system event
  ```
  level=DEBUG ts=2023-03-08T04:36:02.966714Z app=core-metadata source=notify.go:161 msg="Published 
  the 'update' System Event for deviceservice 'device-virtual' to topic 'edgex/system-events/core- 
  metadata/deviceservice/update/device-virtual'"
  ```
But the device-virtual can't receive the system event.

It seems the publish topic does not match the pattern `edgex.system-events.core-metadata.deviceservice.update.device-virtual.*`.

write a test code:
```
func main() {
	redisdb := redis.NewClient(&redis.Options{
		Addr: ":6379",
	})
	pubsub := redisdb.PSubscribe("edgex.system-events.core-metadata.deviceservice.update.device-virtual.*")
	_, err := pubsub.Receive()
	if err != nil {
		panic(err)
	}

	go func() {
		ch := pubsub.Channel()
		for msg := range ch {
			fmt.Println(msg.Channel, msg.Payload)
		}
	}()

	redisdb.Publish("edgex.system-events.core-metadata.deviceservice.update.device-virtual.*", "test topic 1")
	redisdb.Publish("edgex.system-events.core-metadata.deviceservice.update.device-virtual.123", "test topic 2")
	redisdb.Publish("edgex.system-events.core-metadata.deviceservice.update.device-virtual", "test topic 3")

	select {}
}
```
Output
```
edgex.system-events.core-metadata.deviceservice.update.device-virtual.* test topic 1
edgex.system-events.core-metadata.deviceservice.update.device-virtual.123 test topic 2

```

## 🔥 Exception or Error
<pre><code>

</code></pre>

## 🌍  Your Environment
**Deployment Environment:**

**EdgeX Version [**REQUIRED**]:**

**Anything else relevant?**

**Comment by weichou1229** [comment]
@lenny-intel no, this occurs when the service publishes the device service system event.
https://github.com/edgexfoundry/edgex-go/blob/923421b803a2aa1fd35cfe5d85a9354669a43ca8/internal/core/metadata/application/notify.go#L121

For example, 
- the device-service subscribes the system event from the topic 
  `edgex.system-events.core-metadata.*.*.device-virtual.*`
- the core-metadata publishes the system event to the topic 
  `edgex.system-events.core-metadata.deviceservice.update.device-virtual`

The publish topic does not match the subscribe topic pattern in Redis Pub/Sub.

**Comment by cloudxxx8** [comment]
@lenny-intel after some investigation, I believe we should modify go-mod-messaging to subscribe both `abc.edf` and `abc.edf.*` topics if the topic suffix is `.*` to make the behavior consistent with MQTT subscription.
https://github.com/edgexfoundry/go-mod-messaging/blob/71e996eb43c5de4c2fda82be8f8e66c711ad1efe/internal/pkg/redis/goredis.go#L131

Also, the Redis PSubscription accepts multiple channels (topics).
https://github.com/redis/go-redis/blob/2eb36ad9c874df5d87c320a816a3687286d10786/pubsub.go#L227

@jinlinGuan could you please help verify this approach can resolve this issue?

**Comment by jinlinGuan** [comment]
Verified adding the additional topic can fix this issue, PR: https://github.com/edgexfoundry/go-mod-messaging/pull/232

**Comment by cloudxxx8** [comment]
@weichou1229 could you please verify this issue on the latest device-virtual image?

**Comment by weichou1229** [comment]
Verified it works from the edgex-compose repo.
- deploy services `make run no-secty device-virtual`
- send command to lock the device-virtual
```
curl --request PATCH 'http://localhost:59881/api/v2/deviceservice' \
--header 'Content-Type: application/json' \
--data '[
  {
    "apiVersion": "v2",
    "service": {
      "name": "device-virtual",
      "adminState": "LOCKED"
    }
  }
]'
```
- Verify the service is locked
```
$ curl --request POST 'http://localhost:59900/api/v2/discovery' --header 'Content-Type: application/json'
{"apiVersion":"v2","message":"service locked","statusCode":423}
```
